src/components/buttons/accessible-button/accessible-button.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import invariant from 'tiny-invariant';
+import { css } from '@emotion/core';
+import vars from '../../../../materials/custom-properties';
+
+type AccessibleButtonProps = {
+  id?: string;
+  type?: 'submit' | 'reset' | 'button';
+  label: string;
+  children: React.ReactNode;
+  isToggleButton?: boolean;
+  isToggled?: boolean;
+  isDisabled?: boolean;
+  onClick?: (e: React.MouseEvent<any>) => void;
+  className?: string;
+  buttonAttributes?: Object;
+};
+const AccessibleButton: React.FC<AccessibleButtonProps> = React.forwardRef(
+  (props, ref) => {
+    invariant(
+      props.isToggleButton && typeof props.isToggled === 'undefined',
+      'ui-kit/AccessibleButton: when `isToggleButton` is passed `isToggled` is required.'
+    );
+
+    return (
+      <button
+        id={props.id}
+        ref={ref as React.RefObject<HTMLButtonElement>}
```

Comment:
...refs are a bit "annoying" cause you need to state when creating the `ref` what the type of it is. So `createRef<HtmlDivElement | HtmlButtonElement | ...>`. However here we don't know so we just need to cast it. Maybe we can also type the prop and `forwardRef` does the rest.

Reply:
Do you want to try that approach then?

Reply:
Sure. I didn't think it matters too much for what we aim to learn. I found the initial approach less noisy. I changed it in the last commit.